### ISSUE TITLE
Missing setPageLoader(false) in BillingSameAsShippingCheckbox

### ISSUE THREAD

**Issue by DanielRuf** [hyva-themes/magento2-react-checkout#276]
https://github.com/hyva-themes/magento2-react-checkout/blob/main/src/reactapp/src/components/shippingAddress/components/BillingSameAsShippingCheckbox.jsx#L64-L67

It seems it was forgotten to reset the page loader, especially when form validation (field x is required) is triggered again when you check and uncheck the checkbox.

You get a stuck page loader overlay when you toggle the checkbox.
This will fix it:

```jsx
    } catch (error) {
      setPageLoader(false); // this was missing
      console.error(error);
      setErrorMessage(__('Billing address update failed. Please try again.'));
    }
```

See for comparison the other implementations:
https://github.com/hyva-themes/magento2-react-checkout/blob/84b534251d955aba00a3400b9d1d5acb77be536d/src/reactapp/src/components/couponCode/components/CouponCodeForm.jsx#L37
https://github.com/hyva-themes/magento2-react-checkout/blob/71c95fbc8e10acf545910bab7f552b3310f02899/src/reactapp/src/components/paymentMethod/components/PaymentMethodFormManager.jsx#L42

**Comment by rajeev-k-tomy** [comment]
Thank you for reporting the issue. Fix is created and will be available in the next release.
